network-api/networkapi/wagtailpages/pagemodels/products.py

```diff
@@ -426,6 +480,86 @@ def get_context(self, request, *args, **kwargs):
         )} - {self.title}'''
         return context
 
+    def serve(self, request, *args, **kwargs):
+        # In Wagtail we use the serve() method to detect POST submissions.
+        # Alternatively, this could be a routable view.
+        if request.body and request.method == "POST":
```

Comment:
In Wagtail when we want to accept a POST request on a page, we tend to put it in the `serve()` method and check if the request method is POST. 

There is no `def post()` method to overwrite.

Reply:
Might be handy for future readers to link to the docs page on the `serve` function in this comment too,  as it's one of the far-less-used parts of wagtail.